model/modeldecoder/context.go

```diff
@@ -33,65 +32,62 @@ import (
 )
 
 // decodeContext parses all information from input, nested under key context and returns an instance of Context.
-func decodeContext(input interface{}, cfg Config, err error) (*model.Context, error) {
-	if input == nil || err != nil {
-		return nil, err
-	}
-	raw, ok := input.(map[string]interface{})
-	if !ok {
-		return nil, errors.New("invalid type for fetching Context out")
+func decodeContext(input map[string]interface{}, cfg Config) (*model.Context, error) {
```

Comment:
On purpose – the PR was already huge as it is, so I was trying to limit to metadata. I may come back and do that later, otherwise I'd defer to the decoder overhaul.